pkg/ottl/ottlfuncs/func_uuid.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+
+	guuid "github.com/google/uuid"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func uuid[K any]() (ottl.ExprFunc[K], error) {
+	return func(ctx context.Context, tCtx K) (interface{}, error) {
+		u := guuid.New()
+		return u.String(), nil
+	}, nil
+}
+
+func createUuidFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	return uuid[K]()
+}
+
+func NewUuidFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("uuid", nil, createUuidFunction[K])
```

Comment:
Since this is a converter it has to start with a capital letter or the parser won't handle it correctly.  `UUID()` seems like the most appropriate name.

@evan-bradley side note - we should enforce Converter vs Editor in the factory somehow.